src/Native/Runtime/RuntimeInstance.cpp

```diff
@@ -796,10 +796,20 @@ COOP_PINVOKE_HELPER(EEType *, RhGetGenericInstantiation, (EEType *
                                                           EEType ***             ppInstantiation,
                                                           GenericVarianceType ** ppVarianceInfo))
 {
+#if CORERT
+    *pArity = pEEType->get_GenericArity();
+    *ppInstantiation = pEEType->get_GenericArguments();
+    if (pEEType->HasGenericVariance())
```

Comment:
Suggestion, just do pEEType->get_GenericVariance() and have the implementation return NULL when it has no variance mark.

Also adding an ASSERT that pEEType is indeed generic would be useful.